main.go

```diff
@@ -173,12 +175,12 @@ const (
 	online
 )
 
-func (b *Backend) setOffline() {
-	atomic.StoreInt32(&b.up, offline)
+func (b *Backend) setOffline() bool {
+	return atomic.SwapInt32(&b.up, offline) != offline
```

Comment:
ok, so we have `b.Stats.DowntimeStart`, which seems a bit sparsely updated.

Maybe put it in here, so it is updated when offline is flipped. If there is any accumulation of offline time, maybe have that in the setOnline.